### PR TITLE
Set the default for EnableServiceLinks only in Pod

### LINKED ISSUE #69445: New field in Pod spec cause issues with DaemonSet
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

The change in #68754 added a new field (`enableServiceLinks`) to the Pod spec that is set to true by default. This is causing problems for DaemonSets in some situations. 

DaemonSets are using ControllerRevisions to keep track of previous configurations and each revision will contain among other things the Pod spec for the DaemonSet. If a DaemonSet is created with the enableServiceLinks property is set to true, the ControllerRevision will also contain this field. If the cluster is rolled back to a previous version of kubernetes, the DeamonSet struct does not have this field and so the ControllerRevision and the DaemonSet will no longer match. This leads to a new ControllerRevision being created and a rolling update of all pods in the DaemonSet.

Would it be a better solution to rename the field to `disableServiceLinks` and enable service links whenever the field is not set? This would fix the problem with the DaemonSets unless the field is explicitly set before the rollback.

This is causing the master upgrade tests to fail as reported in #69356

/sig apps
/cc @kow3ns
/cc @dchen1107
/cc @thockin

### PR DESCRIPTION
**Pull request by bradhoekstra**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Change how we default the PodSpec.EnableServiceLinks variable. To avoid having the default value materialize in higher level object (DaemonSet, Deployment) and have controllers react to that as an object change only default this value in the actual Pod object.

**Which issue(s) this PR fixes**
Fixes #69445

**Special notes for your reviewer**:
Also related: #69356, #67903

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by bradhoekstra** [comment]
/hold

Waiting on discussion in #69445 that this is the correct approach.

**Comment by smarterclayton** [comment]
Much simpler fix, and agree with the pattern.

/approve

**Comment by mortent** [comment]
This looks good to me, but someone with reviewer status needs to give the lgtm

**Review by mortent — APPROVED** [review]
/lgtm

**Pull request merged**
